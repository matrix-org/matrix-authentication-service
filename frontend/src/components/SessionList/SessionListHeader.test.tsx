// Copyright 2023 The Matrix.org Foundation C.I.C.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// @vitest-environment happy-dom

import { composeStory } from "@storybook/react";
import { render, cleanup } from "@testing-library/react";
import { describe, it, expect, afterEach } from "vitest";

import Meta, { Basic, WithOneAction } from "./SessionListHeader.stories";

describe("<SessionListHeader />", () => {
  afterEach(cleanup);
  it("renders a basic header", () => {
    const Component = composeStory(Basic, Meta);
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });

  it("renders a header with actions", () => {
    const Component = composeStory(WithOneAction, Meta);
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
});
