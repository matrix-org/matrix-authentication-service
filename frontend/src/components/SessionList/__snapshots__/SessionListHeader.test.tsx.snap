// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SessionListHeader /> > renders a basic header 1`] = `
<div>
  <div
    style="width: 378px;"
  >
    <header
      class="_sessionListHeader_5d8138"
    >
      <h5
        class="_font-body-lg-semibold_1jx6b_83 _title_5d8138"
      >
        Apps
      </h5>
    </header>
  </div>
</div>
`;

exports[`<SessionListHeader /> > renders a header with actions 1`] = `
<div>
  <div
    style="width: 378px;"
  >
    <header
      class="_sessionListHeader_5d8138"
    >
      <h5
        class="_font-body-lg-semibold_1jx6b_83 _title_5d8138"
      >
        Apps
      </h5>
      <button
        class="_button_1t47t_17"
        data-kind="destructive"
        data-size="sm"
        role="button"
        tabindex="0"
      >
        End sessions
      </button>
    </header>
  </div>
</div>
`;
